### PR TITLE
Explore: Correct the functionality of shift-enter shortcut across all uses

### LINKED ISSUE #36349: shift-enter shortcut in Explorer logs search doesn't work when 'zoomed' on a time range
**What happened**:
When using Explorer for logs search with Loki (or other sources) the keyboard shortcut shift-enter completes the query and tells grafana to run it. In 8.0.0 and 8.0.3 if you make a search using the default time picker ranges, this works fine. However if you are doing exploratory searching the key shortcut stops functioning. For example, initiate a very wide search like `{app="myapp"}` for a 1 hour range, with lots of results. Then you might decide that a particular window of time has the results you want to look at more closely - if you click and drag the histogram to a range it will 'zoom' to that start/end time range. At this point if you do _any_ modification to the query, for example adding `|= "error"` and use shift-enter keyboard shortcut, grafana does not process they key combination and does not execute the query.

**What you expected to happen**:
Shift-enter keyboard shortcut should work in all query views and time ranges of the Explorer, and it should trigger the query to run.

**How to reproduce it (as minimally and precisely as possible)**:
1. In explorer write a query for logs, something that returns results over several minutes probably
2. Use the 'shift-enter' keyboard shortcut, the query should execute
3. Use the histogram to select a small portion of the results - verify that the time picker shows a specific time range now
4. Edit the logql query; either to a different set of label selectors, or adding a filter expression of any kind that matches some of the log lines in view.
5. Use 'shift-enter' keyboard shortcut, the query will **not** execute 

**Anything else we need to know?**:

**Environment**:
- Grafana version: Tested on 8.0.0 and 8.0.3
- Data source type & version: Loki 2.2.0
- OS Grafana is installed on: Docker/AWS Fargate
- User OS & Browser: MacOS Chrome 
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This is similar issue as we have addressed in ExploreToolbar - https://github.com/grafana/grafana/pull/34297#issuecomment-846928783. When user runs the query with "Run query button" or trough query editor, we want to always run query and reset cache. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/36349

**Special notes for your reviewer**:

How to test this: 
1. In explorer write a query for logs, something that returns results over several minutes probably
2. Use the 'shift-enter' keyboard shortcut, the query should execute
3. Use the histogram to select a small portion of the results - verify that the time picker shows a specific time range now
4. Edit the logql query; either to a different set of label selectors, or adding a filter expression of any kind that matches some of the log lines in view.
5. Use 'shift-enter' keyboard shortcut, the query **WILL** execute

### CODE REVIEW

**Review by ifrost — APPROVED** [review]
Looks good 👍 Would it make sense to make `clearCache` part of `runQueries`?

**Comment by ivanahuckova** [comment]
>Looks good 👍 Would it make sense to make clearCache part of runQueries?

Yeah, that is a very good point. I'll probably do that.

**Comment by ivanahuckova** [comment]
@ifrost I have changed the solution that we always clear the cache in `runQueries` unless we explicitly wan't to keep it and we pass preserveCache param. This param is passed when we run queries trough `UpdateTimeRange` which is what we do in Logs navigation. Let me know what do you think! I like this solution better.

File: public/app/features/explore/QueryRow.test.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ const setup = (propOverrides?: object) => {
     absoluteRange: {} as AbsoluteTimeRange,
     removeQueryRowAction: jest.fn() as any,
     runQueries: jest.fn(),
+    clearCache: jest.fn(),
```
Comment (ifrost) [inline public/app/features/explore/QueryRow.test.tsx:24]: leftover?

**Review by ifrost — APPROVED** [review]

**Pull request merged**